walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,63 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+
 
 public final class FindMeetingQuery {
+  /**Return all possible time intervals for a meeting, taking into account the optional attendees, 
+   if it's possible, otherwise return all the possible time intervals without them.*/
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> meetingsWithOptional = findOptionalTimeForMeeting(events, request, true);
+    if (!meetingsWithOptional.isEmpty()) {
+        return meetingsWithOptional;
+    }
+    return findOptionalTimeForMeeting(events, request, false);
+  }
+
+  /**Return all possible time intervals for a meeting, according to its attendees' events.*/
```

Comment:
Please add spaces after the "/**" and before the final "*/", to make it easier to see the actual comment text.
Also - please document what the withOptional flag does. (Though see my next comment.)